libuavcan/include/uavcan/transport/can_io.hpp

```diff
@@ -36,87 +37,81 @@ struct UAVCAN_EXPORT CanRxFrame : public CanFrame
 #endif
 };
 
-
-class UAVCAN_EXPORT CanTxQueue : Noncopyable
+struct CanTxQueueEntry  // Not required to be packed - fits the block in any case
 {
-public:
-    enum Qos { Volatile, Persistent };
+    MonotonicTime deadline;
+    const CanFrame frame;
+    CanIOFlags flags;
+
+    CanTxQueueEntry(const CanFrame& arg_frame, const MonotonicTime arg_deadline, CanIOFlags arg_flags)
```

Comment:
Why though. Any variable that is not intended to be modified shall be declared `const`, this also applies to function arguments. Again, the codebase is not particularly consistent here, but it's fixable. MISRA 7-1-1, HIC++ 7.1.2.

Reply:
const on a pass-by-value parameter isn't part of the function signature so you aren't declaring anything as const. This is only a MISRA  issue for the function definition.

example:
https://coliru.stacked-crooked.com/a/cb2ae585f9056510

Furthermore, useless const params in declarations are often a sign that the author expected to pass the parameter by reference. For this reason some static analysis will mark this pattern as a defect.

example:
https://coliru.stacked-crooked.com/a/761c38a92659054e

Reply:
> const on a pass-by-value parameter isn't part of the function signature so you aren't declaring anything as const

No no no, it has nothing to do with function signature. It is about mutability of locals: https://coliru.stacked-crooked.com/a/0c75f31daf11ffd1

My point is, the code in question is perfectly correct and the following statement is false:

> const qualifier on pass-by-value argument is useless

Reply:
Stupid github somehow committed my response wrong. I meant to include what you saw above as a mea culpa for incorrectly tagging this as a problem but my lead-in to this part of my response was somehow lost.

You are right, Pavel, because I was incorrectly reviewing this line as a function declaration. It is not. It is a definition. If this were a declaration my argument would be correct. Sorry for the confusion both in my original mistake and my mistakenly abridged reply. 

There is no issue here.

(interestingly enough, I think the abridging is not github's fault but Apple's. I'm having all sorts of fantastic problems with my new mac book's keyboard. You might notice extra  spaces here   and there   in my replies until I get it fixed).